error: #[construct_with] must have at least one argument
  --> $DIR/construct-with-errors.rs:20:7
   |
20 |     #[construct_with()] //~ ERROR #[construct_with] must have at least one argument
   |       ^^^^^^^^^^^^^^

error: #[construct_with] should be of the form #[construct_with(<types>)]
  --> $DIR/construct-with-errors.rs:28:22
   |
28 |     #[construct_with("test")] //~ ERROR #[construct_with] should be of the form #[construct_with(<types>)]
   |                      ^^^^^^
