error: unknown attribute: payload
  --> $DIR/payload_fn2.rs:16:7
   |
16 |     #[payload(length_fn = "length_of_payload")] //~ ERROR: unknown attribute: payload
   |       ^^^^^^^

error[E0412]: cannot find type `PacketWithPayload2Packet` in this scope
  --> $DIR/payload_fn2.rs:20:26
   |
14 | pub struct PacketWithPayload2 {
   | ----------------------------- similarly named struct `PacketWithPayload2` defined here
...
20 | fn length_of_payload(_: &PacketWithPayload2Packet) -> usize { //~ ERROR cannot find type `PacketWithPayload2Packet` in this scope
   |                          ^^^^^^^^^^^^^^^^^^^^^^^^ help: a struct with a similar name exists: `PacketWithPayload2`
